lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1196,7 +1217,12 @@ def __init__(
         self.mode = mode
         if private_key and not private_key.is_valid():
             raise TLSCertificatesError("Invalid private key")
+        if renewal_relative_time <= 0.0 or renewal_relative_time > 1.0:
```

Comment:
I think we should not accept anything below `0.5`, as otherwise we do not use up even half of the available life time of the certificate.